main/src/ca/uwaterloo/flix/api/Bootstrap.scala

```diff
@@ -729,13 +739,50 @@ class Bootstrap(val projectPath: Path, apiKey: Option[String]) {
       * It is up to the caller to set the appropriate options on `flix`.
       * It is often the case that `outputJvm` and `loadClassFiles` must be toggled on or off.
       */
-    def compile(flix: Flix): Validation[CompilationResult, BootstrapError] = {
+    def compile(flix: Flix): Result[CompilationResult, BootstrapError] = {
       flix.compile() match {
-        case Validation.Success(result: CompilationResult) => Validation.Success(result)
-        case Validation.Failure(errors) => Validation.Failure(BootstrapError.GeneralError(flix.mkMessages(errors.toList)))
+        case Validation.Success(result: CompilationResult) => Ok(result)
+        case Validation.Failure(errors) => Err(BootstrapError.GeneralError(flix.mkMessages(errors.toList)))
       }
     }
 
+    /**
+      * Configures `flix` to emit class files to the build directory (on the file system)
+      * in production mode.
+      *
+      * @see [[Bootstrap.getBuildDirectory]]
+      * @see [[Build.Production]]
+      */
+    def configureJarOutput(flix: Flix): Result[Unit, BootstrapError] = {
+      val buildDir = Bootstrap.getBuildDirectory(projectPath)
+      for {
+        _ <- validateDirectory(buildDir)
+      } yield {
+        val newOptions = flix.options.copy(build = Build.Production, outputJvm = true, outputPath = buildDir)
+        flix.setOptions(newOptions)
+        ()
+      }
+    }
+
+    /**
+      * Writes `contents` to the jar file located at `jar`.
+      *
+      * This function also adds a manifest to the jar file.
+      *
+      * Creates the jar file if it does not exist, and truncates it if it already exists.
+      *
+      * @see [[Steps.addManifestToZip]]
+      */
+    def createJar(jar: Path, contents: ZipOutputStream => Unit): Result[Unit, BootstrapError.FileError] = {
```

Comment:
Why does this function need to be higher order?

Reply:
Because of `Using`